src/main/java/sirius/db/mongo/Finder.java

```diff
@@ -379,29 +381,37 @@ public long countIn(Class<?> type) {
      * @return the number of documents found
      */
     public long countIn(String collection) {
-        return countIn(collection, false, 0);
+        return countIn(collection, false, 0).orElse(0L);
     }
 
     /**
      * Counts the number of documents in the result of the given query.
      * <p>
      * Note that limits are ignored for this query.
-     * If the there are no filters in this query and forceAccurate is false, a pre-counted estimate is returned instead.
+     * If there are no filters in this query and forceAccurate is false, a pre-counted estimate is returned instead.
      *
      * @param collection    the collection to search in
      * @param forceAccurate if set to <tt>true</tt> we'll never use <b>estimatedDocumentCount</b> which is way more efficient but might return wrong values in case a cluster is active which had experienced an unclean shutdown.
      * @param maxTimeMS     the maximum process time for this cursor in milliseconds, 0 for unlimited
-     * @return the number of documents found
+     * @return the number of documents found, wrapped in an Optional, or an empty Optional if the query timed out
      */
-    public long countIn(String collection, boolean forceAccurate, long maxTimeMS) {
+    public Optional<Long> countIn(String collection, boolean forceAccurate, long maxTimeMS) {
         Watch w = Watch.start();
         try {
             if (filterObject.isEmpty() && !forceAccurate) {
-                return mongo.db().getCollection(collection).estimatedDocumentCount(new EstimatedDocumentCountOptions().maxTime(maxTimeMS, TimeUnit.MILLISECONDS));
+                return Optional.of(mongo.db()
```

Comment:
Can the Mongo Methods return null? In that case it should be `ofNullable` instead of `of` as it will throw an NPE otherwise.

Reply:
these methods return a primitive long